concordium-std/src/test_infrastructure.rs

```diff
@@ -772,6 +725,16 @@ impl Seek for StateEntryTest {
     }
 }
 
+mod trie;
```

Comment:
I think this is an unusal way to organize modules and is not the way we use generally.

When using nested submodules they are either in the same file, or we use a `mod.rs` file that names the root.